src/main/java/com/pusher/client/example/PrivateEncryptedChannelExampleApp.java

```diff
@@ -12,24 +12,23 @@
 public class PrivateEncryptedChannelExampleApp implements
         ConnectionEventListener, PrivateEncryptedChannelEventListener {
 
-    private String apiKey = "FILL_ME_IN";
+    private String apiKey = "FILL_ME_IN"; // "key" at https://dashboard.pusher.com
     private String channelName = "private-encrypted-channel";
     private String eventName = "my-event";
     private String cluster = "eu";
 
-    private final PrivateEncryptedChannel channel;
+    private PrivateEncryptedChannel channel;
 
     public static void main(final String[] args) {
         new PrivateEncryptedChannelExampleApp(args);
     }
 
     private PrivateEncryptedChannelExampleApp(final String[] args) {
-
-        if (args.length == 3) {
-            apiKey = args[0];
-            channelName = args[1];
-            eventName = args[2];
-            cluster = args[3];
+        switch (args.length) {
```

Comment:
I've not seen this done before for setting a bunch of objects - is this going to go through each item setting it, or just to call it once and set only the last element?

Reply:
Because the `break;` is missing from each case, they will all "fall though". It will start at the matching case and execute all the remaining ones. I've not seen it either, but it does seem quite neat.